src/modules/job-manager/journal.h

```diff
@@ -0,0 +1,46 @@
+
+/************************************************************\
+ * Copyright 2019 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#ifndef _FLUX_JOB_MANAGER_JOURNAL_H
+#define _FLUX_JOB_MANAGER_JOURNAL_H
+
+#include <stdarg.h>
+#include <flux/core.h>
+#include <jansson.h>
+
+#include "job.h"
+#include "job-manager.h"
+
+/* Process the event as needed, such as incrementing job sequence
+ * number, and send to any listeners that request the event.
+ */
+int journal_process_event (struct journal *journal,
+                           struct job *job,
+                           int *seqp,
+                           const char *name,
+                           json_t *entry);
```

Comment:
I'd probably call this something like `journal_append_event()`.

Would it be OK to just pass in the job ID rather than the whole job struct?

I asked about the sequence number in an earlier comment.  If we could nix that it would be nice.

Reply:
> Would it be OK to just pass in the job ID rather than the whole job struct?

Looks that'll work now given all the changes.  Good catch.

Reply:
hmmm, to me `journal_append_event()` suggests we're only appending to the journal list, but the function also sends to listeners.  a better name isn't coming to mind.  I considered `journal_send_event()` but it similarly suggests just sending.  We could of course be dumb and call it `journal_send_and_append_event()`.

Reply:
Well I was thinking that the abstract model is that the journal is a queue that is added to by the job manager and  consumed by users, so from the job manager perspective, "append" would be OK.  For example, in the future we might want to use a prep/check/idle pattern inside journal.c to wake up consumers after new events have been added to the journal.

No big deal though - it's an internal function and I'm fine if we leave it.